dnf/conf/config.py

```diff
@@ -243,6 +245,32 @@ def get_reposdir(self):
             dnf.util.ensure_dir(myrepodir)
         return myrepodir
 
+    def _check_remote_file(self, optname):
+        """
+        In case the option value is a remote URL, download it to the temporary location
+        and use this temporary file instead.
+        """
+        prio = self._get_priority(optname)
+        val = self._get_value(optname)
+        if isinstance(val, basestring):
+            location = urlparse.urlparse(val)
+            if location[0] in ('file', ''):
+                # just strip the file:// prefix
+                self._set_value(optname, location.path, prio)
+            else:
```

Comment:
Shouldn't we check the schema matches `('http', 'https')`? For the rare case a user tries something exotic on us or perhaps has a typo in there? They'd get a URL download error, but a more specific error would be better?

Alternatively, doing the check for the download first and then use the local path as a fallback might be better still?

Reply:
The user will get pretty specific error in this case:
```
$ dnf -c ghth://the.location/file.cfg
Config error: URL "ghth://the.location/file.cfg" could not be downloaded: Curl error (1): Unsupported protocol for ghth://the.location/file.cfg [Protocol "ghth" not supported or disabled in libcurl]
```

Reply:
Fair enough.